### PR TITLE
docs: migrated `predicate` docs snippets

### LINKED ISSUE #2950: Migrate snippets from `guide/predicates`
- [ ] `index.test.ts`
- [ ] `interacting-with-predicates.test.ts`
- [ ] `predicate-custom-transactions.test.ts`
- [ ] `predicate-with-configurable.test.ts`
- [ ] `send-and-spend-funds-from-predicates.test.ts`

### PR DESCRIPTION
**Pull request by petertonysmith94**
- Closes #2950 

# Summary

- Migrated `predicate` docs snippets

# Checklist

- [X] All **changes** are **covered** by **tests** (or not applicable)
- [X] All **changes** are **documented** (or not applicable)
- [X] I **reviewed** the **entire PR** myself (preferably, on GH UI)
- [X] I **described** all **Breaking Changes** (or there's none)

### CODE REVIEW

File: apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md [hunk 1]
```diff
@@ -40,26 +40,20 @@ Once the predicate resolves with a return value `true` based on its predefined c
...
-<<< @/../../docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts#send-and-spend-funds-from-predicates-7{ts:line-numbers}
-
-## Pre-staging a Transaction
-
-In some cases, you may want to pre-stage a predicate transaction before submitting it for execution. To do this, you can use the `createTransfer` method on the `Predicate` class.
-
-In the following example, we are pre-staging a transaction to be able to know the transaction ID without actually submitting the transaction.
-
-<<< @/../../docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts#predicates-prestage-transaction{ts:line-numbers}
+<<< @/../../docs-snippets2/src/predicates/cookbook/failure-returns-false.ts#send-and-spend-funds-from-predicates-7{ts:line-numbers}
```
Comment (arboleya) [inline apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md:59]: Did you remove the "Pre-stating a Transaction" guide section on purpose?

**Review by arboleya — APPROVED** [review]

File: apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md (hunk 1 above)
Comment (petertonysmith94) [reply apps/docs/src/guide/predicates/send-and-spend-funds-from-predicates.md:59]: It was almost identical to the [snippet above](https://github.com/FuelLabs/fuels-ts/pull/3358/files#diff-c91f563a89b2a75ace2424867818c2a79dbcb2fabc073472c3ddfebcc370f79aR41-R45), so I move the relevant information.

File: apps/docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts [hunk 2]
```diff
@@ -195,6 +195,7 @@ describe('Send and Spend Funds from Predicates', () => {
     const amountToReceiver = 200;
 
     // #region send-and-spend-funds-from-predicates-8
+    // Create the transaction for transferring funds from the predicate.
```
Comment (danielbate) [inline apps/docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts:198]: How come changes were made to the original snippet?

**Review by danielbate — APPROVED** [review]

File: apps/docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts (hunk 2 above)
Comment (petertonysmith94) [reply apps/docs-snippets/src/guide/predicates/send-and-spend-funds-from-predicates.test.ts:198]: This file should be deleted - will do this in #3369

**Review by Torres-ssf — APPROVED** [review]

**Pull request merged**